registrations/views.py

```diff
@@ -363,6 +363,66 @@ def get(self, request, *args, **kwargs):
         return Response(resp, status=status)
 
 
+class JembiFacilityCheckHealthcheckView(APIView):
+
+    """ Jembi Facility Check Healthcheck Interaction
+        GET - returns service up - getting auth'd requires DB
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        if not (
+            settings.JEMBI_BASE_URL
+            and settings.JEMBI_USERNAME
+            and settings.JEMBI_PASSWORD
+        ):
+
+            return Response(
+                "Jembi integration is not configured properly.",
+                status.HTTP_503_SERVICE_UNAVAILABLE,
+            )
+
+        try:
+            clinic_code = request.query_params.get("clinic_code")
+            result = requests.get(
+                urljoin(settings.JEMBI_BASE_URL, "NCfacilityCheck"),
+                headers={"Content-Type": "application/json"},
+                auth=(settings.JEMBI_USERNAME, settings.JEMBI_PASSWORD),
+                params={"criteria": "value:{}".format(clinic_code)},
+                verify=False,
+                timeout=10.0,
+            )
+            result.raise_for_status()
+
+            jembi_result = result.json()
+
+            if len(jembi_result.get("rows")) > 0:
+                resp = {
+                    "up": True,
+                    "result": {"Facility": jembi_result.get("rows")[0][2]},
+                }
+            else:
+                raise Http404()
+
+        except (requests.exceptions.HTTPError,) as e:
+            if e.response.status_code == 400:
+                logger.warning(
+                    "400 Error when posting to Jembi.\n"
+                    "Response: %s\nPayload:%s" % (e.response.text)
+                )
+                return Response(
+                    "Error when posting to Jembi. Body: %s Payload: %r"
+                    % (e.response.content),
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+            else:
+                raise e
+                return Response("Timeout", status.HTTP_504_GATEWAY_TIMEOUT)
```

Comment:
I don't think that this line will ever be reached?

Reply:
both line 420 and 421?

Reply:
Just 421. Line 420 will reraise the exception, which means that we'll never be able to get to line 421

Reply:
in your opinion, if I remove line 421 and replace it on line 406, (where the timeout of the request will be), would that make more sense?

Reply:
We probably want to catch the timeout error separately, and return the timeout response in that case. Line 406 means that we've gotten a successful response, so it hasn't timed out.